File: Autos-stock-manager-services/src/main/java/org/autostock/dtos/VoitureDetailDto.java
package org.autostock.dtos;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
public class VoitureDetailDto {
    private Long id;
    private Long idMarque;
    private Long modele;
    private Long idModele;
    private Long idFournisseur;
    private Integer annee;
    private String couleur;
    private String vin;
    private BigDecimal prixAchat;
    private BigDecimal prixVente;
    private String statut;
    private LocalDateTime dateEntreeStock;
    private Long kilometrage;
    private Long owner;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private boolean needsRemark;
}

File: autos-stock-client/src/app/models/VoitureDetailDto.ts
import {StatutVoiture} from "./enums/StatutVoiture";
import {CategorieVoiture} from "./enums/CategorieVoiture";

export interface VoitureDetailDto {
  owner: number;
  id: number;
  idMarque: number;
  idModele: number;
  annee: number;
  prixAchat: number;
  /** Prix affiche sur la vitrine, saisi sur la fiche. */
  prixDemande: number;
  /** Montant reellement encaisse, renseigne par la vente. */
  prixVente: number;
  depenseDivers: number;
  idFournisseur: number;
  vin?: string;
  couleur?: string;
  kilometrage?: number;
  statut: StatutVoiture;
  categorie?: CategorieVoiture;
  categorieLabel?: string;
  createdAt?: string;
  updatedAt?: string;
  dateEntreeStock?: string;
  needsRemark: boolean;
  description?: string;
}
